improve retry documentation and add all options (#246)
php-http/documentation: plugins/retry.rst

Async
-----

This plugin is not fully compatible with asynchronous behavior, as the wait
between retries is done with a blocking call to a sleep function.

Options
-------

``retries``: int (default: 1)

Number of retry attempts to make before giving up.

> ``error_response_decider``: callable (default: retry if status code is in 5xx range)

Reviewer: ```suggestion
``error_response_decider``: callable (default behaviour: retry if status code is in 5xx range)
```

> A callback function that gets the request and response to decide whether the

Reviewer: ```suggestion
A callback function that receives the request and response to decide whether the
```

request should be retried.

> ``exception_decider``: callable (default: retry if not a HttpException or status code is in 5xx range)

Reviewer: ```suggestion
``exception_decider``: callable (default behaviour: retry if the exception is not an HttpException or status code is in 5xx range)
```

> A callback function that gets a request and an exception to decide after a

Reviewer: ```suggestion
A callback function that receives a request and an exception to decide after a
```

failure whether the request should be retried.

> ``error_response_delay``: callable (default: exponential backoff)

Reviewer: ```suggestion
``error_response_delay``: callable (default behaviour: exponential backoff)
```

> A callback that gets a request and response and the current number of retries

Reviewer: ```suggestion
A callback that receives a request, a response, the current number of retries
```

and returns how many microseconds we should wait before trying again.

> ``exception_delay``: callable (default: exponential backoff)

Reviewer: ```suggestion
``exception_delay``: callable (default behaviour: exponential backoff)
```

> A callback that gets a request, an exception and the current number of retries 

Reviewer: ```suggestion
A callback that receives a request, an exception, the current number of retries 
```

and returns how many microseconds we should wait before trying again.

Interaction with Exceptions
---------------------------

> If you use the :doc:`ErrorPlugin <error>`, you should place it later in the

Reviewer: ```suggestion
If you use the :doc:`ErrorPlugin <error>`, you should place it after the RetryPlugin in the
```
